cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake

```diff
@@ -6,70 +6,78 @@ macro(SetupShibokenAndPyside)
+    endif(NOT PYSIDE_INCLUDE_DIR AND TARGET PySide${PYSIDE_MAJOR_VERSION}::pyside${PYSIDE_MAJOR_VERSION})
 
     if(NOT PYSIDE_INCLUDE_DIR)
-        message("==================\n"
-                "PySide2 not found.\n"
-                "==================\n")
-    endif(NOT PYSIDE_INCLUDE_DIR)
+        find_pip_include_dir(PySide${PYSIDE_MAJOR_VERSION} PYSIDE_INCLUDE_DIR)
+    endif()
 
-    find_package(PySide2Tools QUIET) #REQUIRED # PySide2 utilities (pyside2-uic & pyside2-rcc)
-    if(NOT PYSIDE2_TOOLS_FOUND)
+    find_package(PySide${PYSIDE_MAJOR_VERSION}Tools QUIET) # PySide utilities (uic & rcc executables)
+    if(NOT PYSIDE_TOOLS_FOUND)
         message("=======================\n"
-                "PySide2Tools not found.\n"
+                "PySide${PYSIDE_MAJOR_VERSION}Tools not found.\n"
                 "=======================\n")
-    endif(NOT PYSIDE2_TOOLS_FOUND)
+    endif()
 
     file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/Ext/PySide)
     file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/__init__.py "# PySide wrapper\n"
-                                  "from PySide2 import __version__\n"
-                                  "from PySide2 import __version_info__\n")
-    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtCore.py "from PySide2.QtCore import *\n\n"
+                                "from PySide${PYSIDE_MAJOR_VERSION} import __version__\n"
+                                "from PySide${PYSIDE_MAJOR_VERSION} import __version_info__\n")
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtCore.py "from PySide${PYSIDE_MAJOR_VERSION}.QtCore import *\n\n"
                                 "#QCoreApplication.CodecForTr=0\n"
                                 "#QCoreApplication.UnicodeUTF8=1\n")
-    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtGui.py  "from PySide2.QtGui import *\n"
-                                "from PySide2.QtWidgets import *\n"
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtGui.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtGui import *\n"
+                                "from PySide${PYSIDE_MAJOR_VERSION}.QtWidgets import *\n"
                                 "QHeaderView.setResizeMode = QHeaderView.setSectionResizeMode\n")
-    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtSvg.py  "from PySide2.QtSvg import *\n")
-    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtUiTools.py  "from PySide2.QtUiTools import *\n")
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtSvg.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtSvg import *\n")
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtUiTools.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtUiTools import *\n")
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtWidgets.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtWidgets import *\n")
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtNetwork.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtNetwork import *\n")
+    file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtWebEngineWidgets.py  "from PySide${PYSIDE_MAJOR_VERSION}.QtWebEngineWidgets import *\n")
```

Comment:
I think it would be reasonable to not make the GUI-oriented ones if BUILD_GUI is off, but since external workbenches will ultimately be encouraged to use these (instead of hardcoding the PySide version into their addon), I think we should create the whole set. The overhead in doing so is very, very small.

Reply:
OK, that's the direction I went in the latest commit - I don't add the GUI files if BUILD_GUI is off, but if it's on I add them all so that Addons can count on their existence.

Reply:
It's not about the overhead. For build files I always think from the Gentoo perspective, where many dependencies can be configured by the user. 
Suppose a user configuring FreeCAD for a server, i.e. headless, `BUILD_GUI=OFF`. This won't require some of the Qt libraries to be installed on the system. Likewise, pyside doesn't need to have support for these libraries, else the pyside package will pull in these libraries, which are actually not needed. So to be able to respect the users choice, I want to have the build interface clean and not unnecessarily depend on libraries which aren't acutally needed, therefore I try to provide different settings for the pyside dependency, depending on what the user has configured for FreeCAD.

> OK, that's the direction I went in the latest commit - I don't add the GUI files if BUILD_GUI is off, but if it's on I add them all so that Addons can count on their existence.

I support this, for the reasons I told above.